db/view/view.cc

```diff
@@ -1178,6 +1189,23 @@ void view_updates::generate_update(
             // TODO: support static rows in views with pk only including columns from base pk
             return;
         }
+        // Optimization suggested in #20679: Here, the view key has the same
+        // columns as the base key, so the base row simply corresponds to a
+        // view row and we can send the update directly to it after the
+        // appropriate conversions (reordering the columns and converting
+        // unselected columns to empty virtual columns) - and we don't need
+        // "existing" (in fact it might not have been read) or the *_entry()
+        // functions that require it.
+        // However, if for some reason "existing" does exist, it means we did
+        // read it (e.g., because another view exists and it needed the read-
+        // before write), so we might as well fall through to use it - it can
+        // be useful for certain optimizations (see for example the test
+        // boost/view_schema_test test_view_update_generating_writetime which
+        // confirms one of them).
```

Comment:
Can you expand on those optimizations? I tried following the test you named but couldn't understand where the row helps.

Reply:
Yes, I actually added in the patch a comment to that exact test to explain this optimization - but I'll try again here:

Before this patch, we had the following optimization: Consider that we have an _unselected_ column in a view whose key has the same columns as the base. So this column is copied to the view as a "virtual column" (has timestamp but not value). The optimization we had was the following: If an update only modifies unselected columns, and these columns _already_ had a value (so a virtual column already existed in the view) - the view write is not done. This view write was supposed to only update the timestamp, not the value (virtual columns had no values) and it turns out this timestamp in the view is not important.

After this patch, we won't do read-before-write in this case, so we can't know whether a virtual column already existed or not - so we can't avoid the view write. So we gained an optimization (don't do read-before-write) and lost one (we can't avoid the view write that we avoided in the past).

But, this comment is about the "worst of all worlds" situations: Imagine that because of a **different** view on the same table, we had to do a read-before-write. So we can't do the avoid-read-before-write optimization. In this case wouldn't it be a shame to loose the old avoid-write optimization?

So in the code you asked about, if "exists" - which means we did do a read-before-write - we fall back to using the old code, which relies on the "exists" (the read-before-write) and gains the additional optimization (and possibly other optimizations too).